### PR TITLE
feat(executor): improve explain for physical plans

### PR DESCRIPTION
**Pull request by nanderstabel**
Output of:
```
explain select
    l_returnflag,
    l_linestatus,
    sum(l_quantity) as sum_qty,
    sum(l_extendedprice) as sum_base_price,
    sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,
    sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge,
    avg(l_quantity) as avg_qty,
    avg(l_extendedprice) as avg_price,
    avg(l_discount) as avg_disc,
    count(*) as count_order
from
    lineitem
where
    l_shipdate <= date '1998-12-01' - interval '71' day
group by
    l_returnflag,
    l_linestatus
order by
    l_returnflag,
    l_linestatus;
```
now looks like this:
```
PhysicalOrder: [InputRef #0 (asc), InputRef #1 (asc)]
  PhysicalProjection:
      InputRef #0
      InputRef #1
      InputRef #2 (alias to sum_qty)
      InputRef #3 (alias to sum_base_price)
      InputRef #4 (alias to sum_disc_price)
      InputRef #5 (alias to sum_charge)
      (InputRef #6 / InputRef #7) (alias to avg_qty)
      (InputRef #8 / InputRef #9) (alias to avg_price)
      (InputRef #10 / InputRef #11) (alias to avg_disc)
      InputRef #12 (alias to count_order)
    PhysicalHashAgg:
        sum(InputRef #3) -> NUMERIC(15,2)
        sum(InputRef #4) -> NUMERIC(15,2)
        sum((InputRef #4 * (1 - InputRef #5))) -> NUMERIC(15,2) (null)
        sum(((InputRef #4 * (1 - InputRef #5)) * (1 + InputRef #6))) -> NUMERIC(15,2) (null)
        sum(InputRef #3) -> NUMERIC(15,2)
        count(InputRef #3) -> INT
        sum(InputRef #4) -> NUMERIC(15,2)
        count(InputRef #4) -> INT
        sum(InputRef #5) -> NUMERIC(15,2)
        count(InputRef #5) -> INT
        count(InputRef #7) -> INT
      PhysicalTableScan:
          table #7,
          columns [10, 8, 9, 4, 5, 6, 7, 0],
          with_row_handler: false,
          is_sorted: false,
          expr: LtEq(InputRef #0, Date(Date(10490)) (const))
```
* Several plans which previously had a very long output line now have a more visually pleasing multiline output (with correct indentation). 
* ``PhysicalHashAgg``'s aggregation functions are now printed (multiline).
* Some expressions are now more human-readable, example: `` Divide(InputRef #6, InputRef #7 as Decimal(Some(15), Some(2))) (alias to avg_qty)`` is now written as ``(InputRef #6 / InputRef #7) (alias to avg_qty)``.

Considerations:
* My implementations feel a bit hacky and not idiomatic. I feel that I might have unintentionally breached some design principles so please let me know if that's the case.
* Further improvement could be to replace all ``InputRef`` with their actual refered input.
* Write a bunch of tests.
* ...

part of https://github.com/risinglightdb/risinglight/issues/410

### CODE REVIEW

File: src/binder/expression/agg_call.rs [hunk 1]
```diff
@@ -56,6 +56,12 @@ impl std::fmt::Debug for BoundAggCall {
     }
 }
 
+impl std::fmt::Display for BoundAggCall {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}({}) -> {}", self.kind, self.args[0], self.return_type)
```
Comment (skyzh) [inline src/binder/expression/agg_call.rs:61]: Agg call could have zero or >1 args, and therefore we should format and join all args together.

Comment (skyzh) [reply src/binder/expression/agg_call.rs:61]: e.g., count(*) have 0 args, and string_agg (not implemented yet) has 2 args.

**Review by skyzh — COMMENTED** [review]
Rest LGTM, good work!

**Comment by skyzh** [comment]
> My implementations feel a bit hacky and not idiomatic. I feel that I might have unintentionally breached some design principles so please let me know if that's the case.

No problem AFAIK.

> Further improvement could be to replace all InputRef with their actual refered input.

That would be a hard task, as we might need to recursively find the referred item... I think a better way to make the explain output more human-readable, so that developers can easily know the relationship between those columns:

```
PhysicalOrder: [InputRef #0 (asc), InputRef #1 (asc)]
  PhysicalProjection:
      0: InputRef #0
      1: InputRef #1
      2: InputRef #2 (alias to sum_qty)
      3: InputRef #3 (alias to sum_base_price)
      4: InputRef #4 (alias to sum_disc_price)
      5: InputRef #5 (alias to sum_charge)
      6: (InputRef #6 / InputRef #7) (alias to avg_qty)
      7: (InputRef #8 / InputRef #9) (alias to avg_price)
      8: (InputRef #10 / InputRef #11) (alias to avg_disc)
      9: InputRef #12 (alias to count_order)
    PhysicalHashAgg:
        0: sum(InputRef #3) -> NUMERIC(15,2)
        1: sum(InputRef #4) -> NUMERIC(15,2)
        2: sum((InputRef #4 * (1 - InputRef #5))) -> NUMERIC(15,2) (null)
        3: sum(((InputRef #4 * (1 - InputRef #5)) * (1 + InputRef #6))) -> NUMERIC(15,2) (null)
        4: sum(InputRef #3) -> NUMERIC(15,2)
        5: count(InputRef #3) -> INT
        6: sum(InputRef #4) -> NUMERIC(15,2)
        7: count(InputRef #4) -> INT
        8: sum(InputRef #5) -> NUMERIC(15,2)
        9: count(InputRef #5) -> INT
        10: count(InputRef #7) -> INT
      PhysicalTableScan:
          table #7,
          columns [10, 8, 9, 4, 5, 6, 7, 0],
          with_row_handler: false,
          is_sorted: false,
          expr: LtEq(InputRef #0, Date(Date(10490)) (const))
```

Note that for two-input executors like join, we will need to continue the number for the second input instead of starting from 0.

If you'd like to work on this, you may create a separate issue and send another PR. Thanks!

> Write a bunch of tests.

Explain output seems like the UI for a database system, so I don't think we need tests for this.

**Comment by skyzh** [comment]
Hi @nanderstabel, as I'm really looking forward to this feature, I've resolved the comments in this PR by myself and I'll get it merged. Thanks!

**Review by skyzh — APPROVED** [review]

**Comment by nanderstabel** [comment]
@skyzh That's great thank you! I got sick a few days back so I decided to get some rest before fixing the things you commented on. But good that that's not nessecary anymore 👍

**Pull request merged**